src/common/Core/LinuxDesktopEnvironment.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * All supported Linux desktop environments.
+ */
+export const SupportedLinuxDesktopEnvironments = [
+    "GNOME",
+    "GNOME-Classic",
+    "GNOME-Flashback",
+    "KDE",
+    "MATE",
+    "XFCE",
+    "Cinnamon",
+    "X-Cinnamon",
+    "Pantheon",
+] as const;
```

Comment:
In my opinion this list of supported linux desktop environments should not be in `common/Core` one extension or module could support a few of those and another module could support others. I think it's wrong to have a global list of supported environments.

I suggest moving this to the application search extension and to the file image generator module.

Reply:
I had completely overlooked that modules might support different environments. I will move checking support to the individual modules. Could I keep `LinuxDesktopEnvironmentModule` as well as a list of all Linux environments in `common/Core`, supported or not, to have a common type to compare Linux desktops?

Reply:
If it's just a type I'm fine with keeping it. I suggest just using a string literal type:

```typescript
export type LinuxDesktopEnvironment = "GNOME" | "GNOME-Classic" ...
```

Reply:
I added a suggestion here: #1172